### DISCUSSION TITLE
Qt Resource System Internals

### DISCUSSION THREAD

**Discussion by jbruechert** [woboq/qmetaobject-rs#194] [Q&A] [answered]
Hi!

My question isn't directly related to qmetaobject-rs, but I'm asking here because you are probably knowledgeable on the topic.

I'm currently trying to implement support for QRC in the QML bindings for Haskell, similar to how it is implemented in qmetaobject-rs.

However I'm hitting a weird problem: 
Only some ordering of the files works.

This works:
![works](https://user-images.githubusercontent.com/13609393/129235168-94263e11-6239-46dc-a580-88ea526296b3.png)

But this doesn't:
![doesn't work](https://user-images.githubusercontent.com/13609393/129235175-b3875237-c38d-41fe-b053-abf9396e6c4e.png)
(fails with No such file or directory)

Do you know if files need to be placed in the binary format in a certain order?

Just in case you know haskell and want to have a look at the code, you can find it here: https://codeberg.org/jbb/HsQML/pulls/1
I'm not very experienced with Haskell myself, so if you find code that could be expressed better, please let me know.

Thanks!

**Comment by ogoffart** [comment]
As far as i know, we don't sort the files. Maybe we are doing it wrong, or there is something else that goes wrong in your binary format. Maybe something to do with padding bytes?

**Reply by jbruechert** [reply]
Padding bytes seem like a good explanation!
I have mostly implemented it based on what you do in qmetaobject-rs, and didn't do anything specific with padding.

**Reply by jbruechert** [reply]
I have not been able to find the difference between our implementations so far, but I'll go on with diffing the binary output.
Thanks for your help!

**Comment by jbruechert** [comment, marked as answer]
For future reference:
From my testing, the files need to be sorted by their hash in the tree. In qmetaobject-rs's case, this happens automatically, since files are stored in a BTreeMap. In my implementation this was missing.
